content/blog/2019-10-08-remote-read-meets-streaming.md

```diff
@@ -0,0 +1,301 @@
+---
+title: Remote Read Meets Streaming
+created_at: 2019-10-08
+kind: article
+author_name: Bartlomiej Plotka
+---
+
+The new Prometheus version 2.13.0 is available and as always, it includes many fixes and improvements ([CHANGELOG](https://github.com/prometheus/prometheus/blob/release-2.13/CHANGELOG.md)). 
+However, there is one feature that some projects and users were waiting for: [chunked, streamed version of remote read API](https://docs.google.com/document/d/1JqrU3NjM9HoGLSTPYOvR217f5HBKBiJTqikEB9UiJL0/edit#heading=h.3en2gbeew2sa).  
+
+In this article I would like to present a deep dive of what we changed in the remote protocol, why and how to effectively use it.
+
+## Remote APIs
+
+Prometheus since the 1.x version had the ability to interact directly with its storage using [remote API](https://prometheus.io/docs/prometheus/latest/storage/#remote-storage-integrations). 
+
+This API offers two methods: **write** and **read** which allows a 3rd party system to either:
+
+* Receive pushed samples by Prometheus: "Write"
+* Pull samples from Prometheus or to Prometheus: "Read"
+
+![Remote read and write architecture](/assets/blog/2019-10-08/remote_integrations.png)
+
+Both methods are using HTTP with messages encoded with [protobufs](https://github.com/protocolbuffers/protobuf). 
+Additionally, both request and response for both methods are optionally compressed using [snappy](https://github.com/google/snappy).
+
+### Remote Write 
+
+This is the most popular way to replicate Prometheus data into 3rd party system. In this mode, Prometheus streams samples, 
+by periodically sending a batch of those to the given endpoint. 
+
+Remote read was recently improved massively in March with [WAL-based remote read](https://grafana.com/blog/2019/03/25/whats-new-in-prometheus-2.8-wal-based-remote-write/) which
+improved the reliability and resource consumption. It is also worth to note that Write is supported by almost all 3rd 
+party integrations mentioned [here](https://prometheus.io/docs/operating/integrations/#remote-endpoints-and-storage).
+
+### Remote Read
+
+Read method is a bit less popular. It was added in [March 2017](https://github.com/prometheus/prometheus/commit/febed48703b6f82b54b4e1927c53ab6c46257c2f) (server side) and did not change much
```

Comment:
```suggestion
Remote read was added in [March 2017](https://github.com/prometheus/prometheus/commit/febed48703b6f82b54b4e1927c53ab6c46257c2f) (server side) and has not seen significant adoption or development since then.
```